pkg/apis/kops/validation/legacy.go

```diff
@@ -796,6 +797,12 @@ func DeepValidate(c *kops.Cluster, groups []*kops.InstanceGroup, strict bool) er
 			if len(errs) != 0 {
 				return errs[0]
 			}
+		default:
+			for _, x := range groups {
+				if len(x.Spec.Volumes) > 0 {
```

Comment:
Unless I've misread, you don't need to loop again here, just ref `g.Spec.Volumes`

Reply:
fixed :-)